actors/market/src/lib.rs

```diff
@@ -1089,12 +1094,14 @@ fn request_miner_control_addrs(
     rt: &mut impl Runtime,
     miner_id: ActorID,
 ) -> Result<(Address, Address, Vec<Address>), ActorError> {
-    let ret = rt.send(
-        &Address::new_id(miner_id),
-        ext::miner::CONTROL_ADDRESSES_METHOD,
-        None,
-        TokenAmount::zero(),
-    )?;
+    let ret = rt
+        .send(
+            &Address::new_id(miner_id),
+            ext::miner::CONTROL_ADDRESSES_METHOD,
+            None,
+            TokenAmount::zero(),
+        )?
+        .with_context_code(ExitCode::USR_ASSERTION_FAILED, || "return expected".to_string())?;
     let addrs: ext::miner::GetControlAddressesReturnParams = ret.deserialize()?;
```

Comment:
This pattern of expecting non-None return data and then deserializing is repeated a bit – I think worth factoring out somehow.

A free helper function here in actors that wraps up both actions is the basic way, but there might be something slightly better possible too.